### PR TITLE
Prepare v2.4.0 release

### PR DESCRIPTION
**Pull request by Snaipe**
Addresses part of the checklist for #401

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+      with them.
+    * Fix: fixed c++20 incompatibility with the criterion allocator (Ersikan).
+    * Fix: fixed escape of special characters in XML output (Paul Khuat-Duy).
+    * Fix: fixed crash when passing an empty string to cr_log_info (Ersikan).
+    * Fix: fixed compilation errors on macOS (Kare Nuorteva, László Várady).
+    * Misc: various documentation fixes (Oleksii Vilchanskyi, Sam Zaydel, Karim
+      Dridi, Florent Poinsard, Tim Gates, László Várady).
+    * Misc: various updates to dependencies (László Várady).
+    * Misc: various build system fixes (Tomasz Sieprawski, László Várady).
+    * Misc: various fixes to the autotools example (Bruno Belany).
```
Comment (MrAnno) [inline ChangeLog:59]: #301 might be worth mentioning (the `--debug` flag was broken).

File: AUTHORS [hunk 2]
```diff
@@ -1,17 +1,37 @@
...
+* David Carlier <dcarlier@filias.info>
+* Ersikan <julien.philippon@epitech.eu>
+* Ethiraric <florian.sabourin@epitech.eu>
+* Florent Poinsard <florent.poinsard@outlook.fr>
 * Howard "hppritcha" Pritchard <howardp@lanl.gov>
-* kaidowei <kaidowei@users.noreply.github.com>
+* Kare Nuorteva <kare.nuorteva@me.com>
+* Karim DRIDI <seth.holth@gmail.com>
 * Kevin "kevinoid" Locke <kevin@kevinlocke.name>
+* Kevin Locke <kevin@kevinlocke.name>
```
Comment (MrAnno) [inline AUTHORS:22]: This line seems to be a duplicate.

File: AUTHORS [hunk 3]
```diff
@@ -1,17 +1,36 @@
-Maintainer:
+Maintainers:
 
 * Franklin "Snaipe" Mathieu <franklinmathieu@gmail.com>
+* László "MrAnno" Várady <laszlo.varady@balabit.com>
```
Comment (MrAnno) [inline AUTHORS:4]: Wow, thank you for listing me!

Can I use my personal email address (laszlo.varady at protonmail.com)?

Comment (Snaipe) [reply AUTHORS]: Of course :slightly_smiling_face:

**Pull request merged**